File: LEVEL2/프린터.java
import java.util.*;
import java.io.*;

class Solution {
    public class Printer {
        int location;
        int prior;
        
        Printer(int location, int prior) {
            this.location = location;
            this.prior = prior;
        }
    }
    
    public int solution(int[] priorities, int location) {
        int answer = 0;
        Queue<Printer> que = new LinkedList<>();
        
        for(int i = 0; i < priorities.length; i++) {
            que.offer(new Printer(i, priorities[i]));
        }
        
        while(que.size() != 0) {
            boolean chk = false;
            int cur = que.peek().prior;
            
            for(Printer p : que) {
                if(cur < p.prior) {
                    chk = true;
                }
            }
            
            if(chk) {
                que.offer(que.poll());
            }
            
            else {
                if(que.poll().location == location) {
                    answer = priorities.length - que.size();
                }
            }
        }
        
        return answer;
    }
}

File: LEVEL2/프린터.cpp
#include <string>
#include <vector>
#include <queue>
#include <algorithm>
using namespace std;

int solution(vector<int> priorities, int location) {
    int answer = 1;
    queue<pair<int, int>> que;
    priority_queue<int> pq;
    
    for(int i = 0; i < priorities.size(); i++) {
        que.push(make_pair(i, priorities[i]));
        pq.push(priorities[i]);
    }
    
    while(!que.empty()) {
        if(que.front().second == pq.top()) {
            if(que.front().first == location) {
                return answer;
            }
            
            else {
                answer++;
                que.pop();
                pq.pop();
            }
        }
        
        else {
            que.push(que.front());
            que.pop();
        }
    }
    
    return answer;
}